pkg/transformers/namespace.go

```diff
@@ -97,5 +97,49 @@ func (o *namespaceTransformer) Transform(m resmap.ResMap) error {
 		}
 
 	}
+	o.updateClusterRoleBinding(m)
 	return nil
 }
+
+func (o *namespaceTransformer) updateClusterRoleBinding(m resmap.ResMap) {
+	samap := resmap.ResMap{}
+	sagvk := schema.GroupVersionKind{Group: "core", Version: "v1", Kind: "ServiceAccount"}
+	for id := range m {
+		if id.Gvk().String() == sagvk.String() {
+			samap[id] = m[id]
+		}
+	}
+
+	for id := range m {
+		if id.Gvk().Kind != "ClusterRoleBinding" && id.Gvk().Kind != "RoleBinding" {
+			continue
+		}
+		objMap := m[id].UnstructuredContent()
+		subjects := objMap["subjects"].([]interface{})
+		for i := range subjects {
+			subject := subjects[i].(map[string]interface{})
+			kind, foundk := subject["kind"]
+			name, foundn := subject["name"]
+			if !foundk || !foundn || kind.(string) != "ServiceAccount" {
+				continue
+			}
+			if name == "default" {
+				mutateField(subject, []string{"namespace"}, true, func(_ interface{}) (interface{}, error) {
+					return o.namespace, nil
+				})
+				subjects[i] = subject
+			} else {
+				for objId := range samap {
+					if objId.Gvk().Kind == kind.(string) && objId.Name() == name.(string) {
+						mutateField(subject, []string{"namespace"}, true, func(_ interface{}) (interface{}, error) {
+							return o.namespace, nil
+						})
+						subjects[i] = subject
+						break
+					}
+				}
+			}
+		}
```

Comment:
I think the logic above can be simplified in breaking in to two parts (selecting if given subject needs to be transformed (either name "default" or lookup on saMap[name] assuming saMap is a map key'd by name) 
and then calling mutateField() once which is called twice currently.

Or may be I missed something here ?

Reply:
That makes sense. I will update it.

Reply:
This is done.